src/common/color-element.js

```diff
@@ -137,6 +140,38 @@ const Self = class ColorElement extends NudeElement {
 
 		customElements.define(this.tagName, this);
 	}
+
+	/**
+	 * Resolve a color value and cache it.
+	 * @param {string} value Color value to resolve.
+	 * @param {Element} element Element to get computed style from to resolve the color value.
+	 * @returns {Color | null} Resolved color value or null if the value cannot be resolved.
+	 */
+	static resolveColor (value, element) {
+		try {
+			return Self.Color.get(value);
+		}
+		catch {
+			// Color.js can't parse the color value; possibly one of the values we can handle gracefully
+			if (Self.#resolvedColors.has(value)) {
+				return Self.#resolvedColors.get(value);
+			}
+
+			if (!CSS.supports("color", value)) {
+				// Not supported or invalid value
+				return null;
+			}
+
+			// One of the supported color values; resolve and cache it
+			element.style.backgroundColor = value;
+			let color = getComputedStyle(element).backgroundColor;
+			color = Self.Color.get(color);
+			Self.#resolvedColors.set(value, color);
```

Comment:
One issue with this is that now multiple components can share the same `Color` object if they use the same string, and I'm pretty sure we modify it. So perhaps we should cache the resolved string and create a new instance every time.